packages/template-retail-react-app/app/components/store-locator/index.jsx

```diff
@@ -0,0 +1,126 @@
+ */
+
+import React, {useState, useEffect} from 'react'
+import PropTypes from 'prop-types'
+import {useIntl} from 'react-intl'
+import {
+    Modal,
+    ModalBody,
+    ModalCloseButton,
+    ModalContent,
+    useBreakpointValue
+} from '@salesforce/retail-react-app/app/components/shared/ui'
+import StoreLocatorContent from '@salesforce/retail-react-app/app/components/store-locator/store-locator-content'
+import {useSearchStores} from '@salesforce/commerce-sdk-react'
+import {
+    SUPPORTED_STORE_LOCATOR_COUNTRIES,
+    DEFAULT_STORE_LOCATORY_COUNTRY,
+    DEFAULT_STORE_LOCATORY_POSTAL_CODE
+} from '@salesforce/retail-react-app/app/constants'
+
+export const getDefaultSearchStoresParams = (intl) => {
+    var formattedStoreLocatorCountries = SUPPORTED_STORE_LOCATOR_COUNTRIES.map(
+        ({countryCode, countryName}) => {
+            return {countryCode: countryCode, countryName: intl.formatMessage(countryName)}
+        }
+    )
+    var defaultCountryCode = formattedStoreLocatorCountries.find(
+        (obj) => obj.countryName == intl.formatMessage(DEFAULT_STORE_LOCATORY_COUNTRY)
+    ).countryCode
+
+    var defaultSearchStoresParams = {
+        countryCode: defaultCountryCode,
+        postalCode: DEFAULT_STORE_LOCATORY_POSTAL_CODE
+    }
+
+    return defaultSearchStoresParams
+}
+
+const StoreLocatorModal = (props) => {
+    const {isOpen, setIsOpen} = props
+    const intl = useIntl()
+    var defaultSearchStoresParams = getDefaultSearchStoresParams(intl)
+
+    const [searchStoresParams, setSearchStoresParams] = useState(defaultSearchStoresParams)
+
+    var searchStoresData = useSearchStores({
+        parameters: {
+            countryCode: searchStoresParams.countryCode,
+            postalCode: searchStoresParams.postalCode,
+            locale: intl.locale,
+            maxDistance: 100
+        }
+    })
+
+    const isDesktopView = useBreakpointValue({base: false, lg: true})
+
+    return (
+        <>
+            {isDesktopView ? (
+                <Modal size="4xl" isOpen={isOpen}>
+                    <ModalContent
+                        display={{base: 'none', lg: 'block'}}
+                        position="absolute"
+                        top="0"
+                        right="0"
+                        width="33.33%"
+                        height="100vh"
+                        borderLeft="1px solid"
+                        borderColor="gray.200"
+                        marginTop="0px"
+                    >
+                        <ModalCloseButton
+                            onClick={() => {
+                                setIsOpen(false)
+                            }}
+                        />
+                        <ModalBody pb={8} bg="white" paddingBottom={6} marginTop={6}>
+                            <StoreLocatorContent
+                                searchStoresData={searchStoresData}
+                                searchStoresParams={searchStoresParams}
```

Comment:
I think you are getting close here. The pattern you've implemented with searchStoresParams and setSearchStoresParams seems like a re-implementation of the pattern established by react-hook-form's useForm() hook. 

We used this pattern of exposing the "onSumbit" handler for the form, which I feel is a lot like what you did with "setSearchSotresParams". 

Take a look at these doc's here --> https://www.react-hook-form.com/api/useform/

See if you can take a look at exposing the onSubmit in your form component, you might want to look at the defaultValues or values to preserver state for your form.

Reply:
The reason for using states here is to be able to call the useSearchStores hook when the user submits an input somewhere in the child component. Rules with hooks prevent this from being called within an event listener, so states were used to re-render the parent component to retrieve the list of stores. Unlike other components that use the `onSubmit` which can pull a function out of the hook because it does an async mutation within the listener, the useSearchStores is a hook to be used by itself.

I also do use the `useForm` in [here](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1827/files#diff-f98827771c19aeabddc19882b03130c1bebc5113e67fcbe6136e310e76050babR28-R36), and I can look to pull it out to the parent component so that the "form" gets passed in to use a form component's submit handler, but I'm not sure if replacing aforementioned use case of the `searchStoresParams` state entirely would work